rasa/shared/core/domain.py

```diff
@@ -619,6 +627,41 @@ def _initialize_forms(
             `FormAction` which is implemented in the Rasa SDK.
         """
         if isinstance(forms, dict):
+            # Updating the `not_intent` parameter of each slot with `global_not_intent`
+            if session_config.global_not_intent is not None:
+                global_params = session_config.global_not_intent
+                if not isinstance(global_params, list):
+                    global_params = [global_params]
+                for global_value in global_params:
+                    # looping through all the available forms, their slots and their
+                    # all their slot's types and updating the `not_intent` key
+                    # with the `global_not_intent` value.
+                    for form_count, _ in enumerate(list(forms.keys())):
+                        form_slots = list(forms.values())[form_count]
+                        for _, slot_name in enumerate(form_slots):
+                            slot_params = form_slots[slot_name]
+                            for type_count, type_name in enumerate(slot_params):
+                                key = "not_intent"
+                                # check that `not_intent` param is present
+                                if key in slot_params[type_count]:
+                                    # check that the value of `not_intent` is a list
+                                    if isinstance(slot_params[type_count][key], list):
+                                        # check that global_not_intent is in the list
+                                        if (
+                                            global_value
+                                            not in slot_params[type_count][key]
+                                        ):
+                                            slot_params[type_count][key].append(
+                                                global_value
+                                            )
+                                    else:
+                                        if global_value != slot_params[type_count][key]:
+                                            slot_params[type_count][key] = [
+                                                slot_params[type_count][key],
```

Comment:
you reference `lot_params[type_count][key]` many times, so may be worth extracting into a variable.